File: src/main/java/me/reckter/telegram/model/update/Update.java
package me.reckter.telegram.model.update;

import com.fasterxml.jackson.annotation.JsonProperty;
import me.reckter.telegram.model.BaseModel;
import me.reckter.telegram.model.Message;

/**
 * @author hannes
 */
public class Update extends BaseModel {

    @JsonProperty("update_id")
    public int id;

    public Message message;

    @JsonProperty("edited_message")
    public Message editedMessage;

    @JsonProperty("callback_query")
    public CallbackQuery callbackQuery;

    public Message getEditedMessage() {
        return editedMessage;
    }

    public void setEditedMessage(Message editedMessage) {
        this.editedMessage = editedMessage;
    }

    public CallbackQuery getCallbackQuery() {
        return callbackQuery;
    }

    public void setCallbackQuery(CallbackQuery callbackQuery) {
        this.callbackQuery = callbackQuery;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Message getMessage() {
        return message;
    }

    public void setMessage(Message message) {
        this.message = message;
    }


}

File: src/main/java/me/reckter/telegram/model/update/Update.kt
package me.reckter.telegram.model.update

import com.fasterxml.jackson.annotation.JsonProperty
import me.reckter.telegram.model.BaseModel
import me.reckter.telegram.model.InlineQuery
import me.reckter.telegram.model.Message

/**
 * @author hannes
 */
class Update : BaseModel() {

    @JsonProperty("update_id")
    var id: Int = 0

    var message: Message? = null

    @JsonProperty("edited_message")
    var editedMessage: Message? = null


    @JsonProperty("callback_query")
    var callbackQuery: CallbackQuery? = null

    @JsonProperty("inline_query")
    var inlineQuery: InlineQuery? = null

    @JsonProperty("chosen_inline_result")
    var chosenInlineResult: ChosenInlineResult? = null

}
